### PR TITLE
Add activity log

### PR DESCRIPTION
**Pull request by paullinator**
#### PR Dependencies

 none

#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [x] Tested on iOS device
- [ ] Tested on Android device
- [x] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202506473095599

### CODE REVIEW

File: src/actions/LogActions.js [hunk 1]
```diff
@@ -35,16 +35,24 @@ export const showSendLogsModal = () => async (dispatch: Dispatch, getState: GetS
 }
 
 const prepareLogs = (text: string) => async (dispatch: Dispatch, getState: GetState) => {
+  const id = Math.floor(Math.random() * 16777215).toString(16)
```
Comment (thehobbit85) [inline src/actions/LogActions.js:38]: Can you use '0xFFFFFF' instead of decimal? It looks less "magic" in hex ;)

File: wyrekey.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+function makeSecret(length) {
```
Comment (thehobbit85) [inline wyrekey.js:1]: What's that? Where is it used?

File: src/actions/CryptoExchangeActions.js [hunk 3]
```diff
@@ -320,12 +320,32 @@ export const shiftCryptoCurrency = (swapInfo: GuiSwapInfo, onApprove: () => void
...
-  const { pluginId, toNativeAmount } = quote
+  const { fromDisplayAmount, quote, request, fee, fromFiat, fromTotalFiat, toDisplayAmount, toFiat } = swapInfo
+  const { isEstimate, fromNativeAmount, toNativeAmount, networkFee, pluginId, expirationDate } = quote
   const { fromWallet, toWallet, fromCurrencyCode, toCurrencyCode } = request
   try {
     logEvent('SwapStart')
     const result: EdgeSwapResult = await quote.approve()
+
+    global.logActivity(`Swap Exchange Executed: ${account.username}`)
+    global.logActivity(`
```
Comment (thehobbit85) [inline src/actions/CryptoExchangeActions.js:331]: Why not pass the information as an object to the logs?

File: src/util/logger.js [hunk 4]
```diff
@@ -78,45 +107,43 @@ async function writeLog(content) {
...
-      return writeLog(d + ': ' + logs)
+      return writeLog(type, d + ': ' + logs)
     })
   } catch (e) {
     global.clog(e)
   }
   global.clog(logs)
 }
 
+export async function log(...info: Array<number | string | null | {}>): Promise<void> {
```
Comment (thehobbit85) [inline src/util/logger.js:144]: Why not create a type `Log: { type: LogType, data: any }` instead of wrapper functions?

**Review by thehobbit85 — CHANGES_REQUESTED** [review]

File: src/actions/CryptoExchangeActions.js (hunk 3 above)
Comment (paullinator) [reply src/actions/CryptoExchangeActions.js:331]: Visual formatting would suck. Also I don't want all object info to be printed as there may be sensitive info in the object so I would have to filter out some items anyway.

File: src/util/logger.js (hunk 4 above)
Comment (paullinator) [reply src/util/logger.js:144]: I'm just trying to give the equivalent of what we do now with console.log but for the activity log. Less for people to "get used to" when trying to use `logActivity`. Any old `console.log` can also easily be ported to use `logActivity`.

File: wyrekey.js (hunk 2 above)
Comment (paullinator) [reply wyrekey.js:1]: whoops. this snuck in the PR. i'll remove

File: src/actions/LogActions.js (hunk 1 above)
Comment (paullinator) [reply src/actions/LogActions.js:38]: okaaaay

Comment (paullinator) [reply src/actions/LogActions.js:38]: Great. now VScode thinks it's a color and colorizes the text 😏

**Review by thehobbit85 — APPROVED** [review]

**Pull request merged**